src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -781,41 +781,43 @@ impl<'tcx> Tree {
                 ProvenanceExtra::Concrete(tag) => self.tag_mapping.get(&tag).unwrap(),
                 ProvenanceExtra::Wildcard => self.root,
             };
-            TreeVisitor { nodes: &mut self.nodes, loc }.traverse_this_parents_children_other(
-                start_idx,
-                // Visit all children, skipping none.
-                |_| ContinueTraversal::Recurse,
-                |args: NodeAppArgs<'_>| {
-                    let node = args.nodes.get(args.idx).unwrap();
-                    let perm = args.loc.perms.entry(args.idx);
+            let check_strong_protector = |args: NodeAppArgs<'_>| {
```

Comment:
Why are you let-binding this closure?

Reply:
See https://github.com/rust-lang/miri/pull/4707#discussion_r2559382962 -- I think there's a better way to share the code here. Probably best to leave that to the other PR.